packages/core/src/login/webview/vue/amazonq/backend_amazonq.ts

```diff
@@ -117,6 +123,56 @@ export class AmazonQLoginWebview extends CommonAuthWebview {
         })
     }
 
+    async reauthenticateConnection(): Promise<void> {
+        this.isReauthenticating = true
+        this.reauthError = undefined
+
+        const error = await this.ssoSetup('reauthenticate', async () => AuthUtil.instance.reauthenticate(true))
+
+        this.isReauthenticating = false
+        this.reauthError = error
+
+        if (error && error.id === userCancelled) {
+            // Since reauth was not successful it did not trigger the switch to Chat.
+            // We must force the auth page to refresh.
+            this.onActiveConnectionModified.fire()
+        }
+    }
+
+    private reauthError: AuthError | undefined = undefined
+    override async getReauthError(): Promise<AuthError | undefined> {
+        return this.reauthError
+    }
+
+    async getActiveConnection(): Promise<Connection | undefined> {
+        return AuthUtil.instance.conn
+    }
+
+    isReauthenticating: boolean = false
+    private authState: AuthFlowState = 'LOGIN'
+    override async updateAuthState(): Promise<void> {
+        const featureAuthStates = await AuthUtil.instance.getChatAuthState()
+        if (featureAuthStates.amazonQ === 'expired') {
+            this.authState = this.isReauthenticating ? 'REAUTHENTICATING' : 'REAUTHNEEDED'
+            return
+        }
+        this.authState = 'LOGIN'
+    }
+
+    override async getAuthState(): Promise<AuthFlowState> {
+        return this.authState
+    }
+
+    override async signout(): Promise<void> {
+        this.resetState()
+        await AuthUtil.instance.secondaryAuth.deleteConnection()
+    }
+
+    private resetState() {
```

Comment:
I've removed `resetState()` since it causes confusion. We already have `isReauthenticating = false` in `reauthenticateConnection()`. But now I've explicitly added it to `signout()`